src/components/BudgetKeyFooterComponent.ts

```diff
@@ -1,14 +1,13 @@
 import {Component, Inject} from '@angular/core';
 import {THEME_TOKEN} from '../constants';
-const Smooch: any = require('smooch');
 
 @Component({
     selector: 'budgetkey-footer',
     template: `
         <footer class="footer">
             <div class="links">
                 <div class="link" *ngFor="let link of theme.footerLinks">
-                    <a [href]="link.href" [innerHtml]="link.title"></a>
+                    <a i18n="@@footerlinks" [href]="link.href" [innerHtml]="link.title"></a>
```

Comment:
Yes, this probably won't work - as you'd get the same translation for all links.

You should replace `link.title` here with a function that translates it into the target language.
For example, if we use the `translations` object we discussed (mapping "id" to "translated text"), we could write here:
`translations[link.title] || link.title` - which will use the value in that object, and in case it doesn't exist, will fallback to `link.title` (`||` is the logical OR operator here).